### PR TITLE
Purchase indicator

### PR DESCRIPTION
**Pull request by EthanVieira**
* added animation for purchasing things
* fixed bug where loaded item from test data wouldn't function properly if that item already existed in inventory

### CODE REVIEW

File: src/data/test-data.js [hunk 1]
```diff
@@ -1,4 +1,16 @@
...
+// ---------------------------------------------------------------------
+// There is a bug where newly added items via altering the saved data won't
+// behave correctly if that item already exists in the inventory. This is normally
+// handled with the addToInventory method, but we bypass that by adding directly to
+// the saved data.
+// To avoid that, update the count of the item instead of pushing a new one.
+// ---------------------------------------------------------------------
+const richData = JSON.parse(JSON.stringify(defaultData));
+
+richData["inventory"].filter((x) => x.item == "Coin")[0].count =
```
Comment (CurtisGreen) [inline src/data/test-data.js]:
```suggestion
richData["inventory"].find((x) => x.item == "Coin").count =
```

File: src/data/test-data.js [hunk 2]
```diff
@@ -1,4 +1,16 @@
...
+import { CONSTANTS } from "../constants/constants.js";
+// ---------------------------------------------------------------------
+// There is a bug where newly added items via altering the saved data won't
+// behave correctly if that item already exists in the inventory. This is normally
+// handled with the addToInventory method, but we bypass that by adding directly to
+// the saved data.
+// To avoid that, update the count of the item instead of pushing a new one.
+// ---------------------------------------------------------------------
+const richData = JSON.parse(JSON.stringify(defaultData));
+
```
Comment (CurtisGreen) [inline src/data/test-data.js:6]:
```suggestion
// Gold already exists in defaultData so we should overwrite it instead of pushing a duplicate stack
```

Comment (EthanVieira) [reply src/data/test-data.js:6]: Updated https://github.com/EthanVieira/rs-clicker/pull/83/commits/a14defe1c3ccc9a77fc775f5da050cff21f46037

File: src/data/test-data.js (hunk 1 above)
Comment (EthanVieira) [reply src/data/test-data.js]: Updated https://github.com/EthanVieira/rs-clicker/pull/83/commits/a14defe1c3ccc9a77fc775f5da050cff21f46037

**Review by CurtisGreen — APPROVED** [review]

**Pull request merged**